### ISSUE TITLE
Add date picker

### ISSUE THREAD

**Issue by dword-design-honestly** [bootstrap-vue-next/bootstrap-vue-next#1860]
### Clear and concise description of the problem

BootstrapVue has a [date picker component](https://bootstrap-vue.org/docs/components/form-datepicker) with styling integrated into the Bootstrap theme. Would be great to have this guy also in bootstrap-vue-next.

### Suggested solution

Migrate the date picker to this project.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by xvaara** [comment]
imho. We already have type="datetime-local" so we don't need a picker anymore in the main libary.

I have bootstrap style date/time -pickers in one client project, I could maybe clean that up for us.
@VividLemon thoughts?

**Comment by dword-design-honestly** [comment]
@xvaara afaik the dropdowns system date inputs aren't styleable because they depend on the OS.

**Comment by xvaara** [comment]
@dword-design-honestly yes, that's why they work universally, and on mobile I prefer the native widget over any other picker.

**Comment by VividLemon** [comment]
Also the bootstrap vue versions appearance is quite dated looking. The native variants simplify things

**Comment by VividLemon** [comment]
> imho. We already have type="datetime-local" so we don't need a picker anymore in the main libary.
> 
> I have bootstrap style date/time -pickers in one client project, I could maybe clean that up for us. @VividLemon thoughts?

I have no opinion. It's your call.

**Comment by dword-design-honestly** [comment]
@VividLemon Would not reuse the styling from BootstrapVue probably, doesn't look that good. Having the option to have a custom widget makes sense to me since, although having the universal one is a pro because it just works, if you have a specific design, it sometimes doesn't integrate well (color of selected date, border radius, padding and stuff).

**Comment by xvaara** [comment]
@VividLemon imho, I'm two minds about this. I'd like to keep the core components simple. Maybe have another repo for more complex components that aren't "bootstrap"... but we already have BTable that's quite complex and "violates that divide"

I'd like us not to have to "create css" only use what bootstrap has provided.

I'd probably just take https://www.radix-vue.com/components/date-picker.html and style that to match my site.


on the other hand, multiselect and date/time picker, better number input, range selector, range on datetime, pin input, color picker, tree ... I've done all of those for client sites that use bootstrap-vue-next and they are nice to have :)

**Comment by VividLemon** [comment]
> but we already have BTable that's quite complex and "violates that divide"

There are places where it does seem appropriate to take over. Tables can have a lot of nested html making it difficult to create many tables. This isn't something that is related to bootstrap, nor is it something that they could fix given their limited knowledge of the app. 

However, other components like spinbutton, date picker, etc are all really bootstrap components. These things could be implemented in bootstrap (within reason, a single "spinbutton" is really just a collection of multiple buttons... So a bootstrap component seems odd for this)

It's a complicated issue. Realistically the only downside is that there is more maintenance on our end. Which is why strategies of abstraction are the best imo, such as embedding radix Vue. Although we have talked about this before. If we were able to abstract things like pin input, date-picker, to an external source, I would be significantly more comfortable with adding many more "standard" components. I'd be fine with rewriting existing components, but I recall you having issues with that idea @xvaara

**Comment by dword-design-honestly** [comment]
Throwing in the argument that most UI frameworks have a themeable date picker component (in contrast to more special components e.g. like a WYSIWYG editor). Ofc. the difficulty is to set the boundary where the framework ends and contributed packages with special components consuming Bootstrap variables start.

I also think that reusing a Vue 3 date picker component and then styling it with Bootstrap makes more sense than reimplementing it.

**Comment by VividLemon** [comment]
If the implementation is a wrapper around https://www.radix-vue.com/components/date-picker , then I'm going to "okay" it

**Comment by EagleTsui** [comment]
BootstrapVue [date picker](https://bootstrap-vue.org/docs/components/form-datepicker) can display with custom format.

The BFormInput only display value as 05/06/2024
The DatePicker can display value as Jun 05, 2024

For Datepicker I can easy to know is 05-Jun-2024, but BFormInput cannot.  the value can 05-Jun-2024 or 06-May-2024, it depend on country.

**Comment by EagleTsui** [comment]
Finally, I found this component instead https://vue3datepicker.com

**Comment by VividLemon** [comment]
https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1871#issuecomment-5329307200
